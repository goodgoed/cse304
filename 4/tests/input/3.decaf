class StaticExample {
  static int counter;
  
  static int incrementCounter() {
    this.counter++;
    return this.counter;
  }
}

class A {
  int incrementCounter(){
    StaticExample.counter++;
    return StaticExample.counter;
  }
}