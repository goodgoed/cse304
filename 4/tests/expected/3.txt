--------------------------------------------------------------------------
Class Name: StaticExample
Superclass Name: 
Fields:
FIELD: 1, counter, StaticExample, private, static, int
Constructors:
Methods:
METHOD: 1, incrementCounter, StaticExample, private, static, int
Method Parameters: 
Variable Table:
Method Body:
Block([
Expr(Auto(Field-access(This, counter, 1), inc, post)),
Return(Field-access(This, counter, 1))
])
--------------------------------------------------------------------------
Class Name: A
Superclass Name: 
Fields:
Constructors:
Methods:
METHOD: 2, incrementCounter, A, private, instance, int
Method Parameters: 
Variable Table:
Method Body:
Block([
Expr(Auto(Field-access(Class(StaticExample), counter, 1), inc, post)),
Return(Field-access(Class(StaticExample), counter, 1))
])
--------------------------------------------------------------------------
